docs/DeveloperGuide.md

```diff
@@ -649,7 +649,51 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 * **Teaching Assistant (TA)**: A tutor attached to class
 
 --------------------------------------------------------------------------------------------------------------------
+## **Appendix: Planned Enhancements**
 
+The following are planned enhancements that have been proposed by early testers to improve the user experience.
+
+Team size: 5
+
+### Further validation of student details
+
+Some of the properties of a `Student` can be further restricted to be more representative of their real-life counterparts.
+
+* Currently, the `Phone` property is a string with unlimited length. It should have a limit of 8 to 15 characters, since some international phone numbers can go up to 15 characters.
+  * Phone numbers in Singapore should start with `8` or `9`.
+* `Email` addresses should follow the _local_@_domain_.com format.
+* Students should have unique contact details.
+  * This means that `add` or `edit` command will fail if it results in 
+* Student `Name`s should not be case sensitive.
+  * This means that `UniqueStudentList` should treat `Tan Ah Beng` and `tan ah beng` as the same student and these two names will never exist simultaneously
+  * One possible implementation is to store `Name` as a lowercase string regardless of how the user enters it, e.g. entering `add n/Tan Ah Beng` will reduce the name to `tan ah beng` before adding it to the list.
+
+Cons: Such additional constraints may restrict the user too much 
+
+**Alternative 1**
+* Have as toggleable 'Strict Mode' option that the user can enable of they want ClassMonitor to ensure that they cannot enter fields that do not fulfil the limitations.
+* Otherwise, only limit fields by string length in 'Unstrict Mode'.
+
+### Enhanced Name Input
```

Comment:
Capitalization for heading is inconsistent here, maybe we can standardize across headers.